src/Moq/MethodCall.cs

```diff
@@ -104,16 +104,25 @@ public override void Execute(Invocation invocation)
 
 			this.callbackResponse?.Invoke(invocation.Arguments);
 
-			if ((this.flags & Flags.CallBase) != 0)
+			var methodIsNonVoid = (this.flags & Flags.MethodIsNonVoid) != 0;
+
+			if (!methodIsNonVoid)
 			{
-				invocation.ReturnBase();
+				if ((this.flags & Flags.CallBase) != 0)
+				{
+					invocation.ReturnBase();
+				}
+				else
+				{
+					invocation.Return();
+				}
 			}
 
 			this.raiseEventResponse?.RespondTo(invocation);
```

Comment:
I am not sure I understand the changes you're making to the `Execute` method (the ones both above and below this line of code). Could you please explain them to me?

I find your changes puzzling because they are phrased entirely in terms of the invoked method being `void` or non-`void` (which is a distinction far too coarse to distinguish between event accessor methods and everything else), and because they do not touch this line, which is the only one directly related to event handling.

Is your change therefore related to this PR's main focus at all, or are they an unrelated bugfix?

Reply:
This is indeed unrelated. If I am not mistaken the invocation won't be returned it is: `void`, `callbase` and **matched**. If it isn't matched the pipeline would proceed and the invocation will be returned. Not sure if this could lead to any problem but as far as I understand the idea is to return invocation as soon as possible to instruct the dynamic proxy to GC it.
This can be ignored, I can completely revert this from this PR to avoid confusion.

Reply:
Your analysis appears to be correct, good catch!

Would it be a big hassle to revert this change here, and submit it as a separate PR?

Reply:
Not at all, I will revert this and send this out in the new PR.